libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -1569,10 +1570,41 @@ internal object MapboxRouteLineUtils {
     internal fun getAlternativeRouteDeviationOffsets(
         metadata: AlternativeRouteMetadata
     ): Double {
-        return distinctGranularDistancesProvider(metadata.navigationRoute)?.let {
+        return distinctGranularDistancesProvider(metadata.navigationRoute)?.let { distances ->
+            if (distances.distancesArray.isEmpty() || distances.distance <= 0) {
+                logW(
+                    "Remaining distances array size is ${distances.distancesArray.size} " +
```

Comment:
Is there an NN ticket for investigation? It's a workaround, I guess it can hide some logic inconsistency. If there can be an index which is out of bounds, maybe the whole (or at least some part of) `AlternativeRouteMetadata` is just wrong. As far as I understand, it's a temporary solution to avoid the crash.

Reply:
UPD: I see that it's probably a NAVAPI issue.